File: php/peminjaman.php
<?php
require 'connection.php';

// GET all peminjaman
if ($_SERVER['REQUEST_METHOD'] === 'GET') {
    $sql = "SELECT p.*, a.nama as nama_anggota, b.judul as judul_buku 
            FROM peminjaman p 
            JOIN anggota a ON p.anggota_id = a.id 
            JOIN buku b ON p.buku_id = b.id";
    $result = mysqli_query($conn, $sql);
    $data = array();
    
    while ($row = mysqli_fetch_assoc($result)) {
        $data[] = $row;
    }
    echo json_encode($data);
}

// POST new peminjaman
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $data = json_decode(file_get_contents('php://input'), true);
    
    $tanggal_pinjam = $data['tanggal_pinjam'];
    $tanggal_kembali = $data['tanggal_kembali'];
    $anggota_id = $data['anggota_id'];
    $buku_id = $data['buku_id'];
    
    $sql = "INSERT INTO peminjaman (tanggal_pinjam, tanggal_kembali, anggota_id, buku_id) 
            VALUES ('$tanggal_pinjam', '$tanggal_kembali', $anggota_id, $buku_id)";
    
    if (mysqli_query($conn, $sql)) {
        echo json_encode(["message" => "Peminjaman berhasil ditambahkan"]);
    } else {
        echo json_encode(["message" => "Error: " . mysqli_error($conn)]);
    }
}

// PUT/UPDATE peminjaman
if ($_SERVER['REQUEST_METHOD'] === 'PUT') {
    $data = json_decode(file_get_contents('php://input'), true);
    
    $id = $data['id'];
    $tanggal_pinjam = $data['tanggal_pinjam'];
    $tanggal_kembali = $data['tanggal_kembali'];
    $anggota_id = $data['anggota_id'];
    $buku_id = $data['buku_id'];
    
    $sql = "UPDATE peminjaman SET 
            tanggal_pinjam='$tanggal_pinjam', 
            tanggal_kembali='$tanggal_kembali', 
            anggota_id=$anggota_id, 
            buku_id=$buku_id 
            WHERE id=$id";
    
    if (mysqli_query($conn, $sql)) {
        echo json_encode(["message" => "Peminjaman berhasil diupdate"]);
    } else {
        echo json_encode(["message" => "Error: " . mysqli_error($conn)]);
    }
}

// DELETE peminjaman
if ($_SERVER['REQUEST_METHOD'] === 'DELETE') {
    $id = $_GET['id'];
    
    $sql = "DELETE FROM peminjaman WHERE id=$id";
    
    if (mysqli_query($conn, $sql)) {
        echo json_encode(["message" => "Peminjaman berhasil dihapus"]);
    } else {
        echo json_encode(["message" => "Error: " . mysqli_error($conn)]);
    }
}
?> 

File: lib/models/peminjaman.dart
class Peminjaman {
  final int? id;
  final String tanggalPinjam;
  final String tanggalKembali;
  final int anggotaId;
  final int bukuId;
  final String? namaAnggota;  // Untuk tampilan
  final String? judulBuku;    // Untuk tampilan

  Peminjaman({
    this.id,
    required this.tanggalPinjam,
    required this.tanggalKembali,
    required this.anggotaId,
    required this.bukuId,
    this.namaAnggota,
    this.judulBuku,
  });

  factory Peminjaman.fromJson(Map<String, dynamic> json) {
    return Peminjaman(
      id: int.parse(json['id'].toString()),
      tanggalPinjam: json['tanggal_pinjam'],
      tanggalKembali: json['tanggal_kembali'],
      anggotaId: int.parse(json['anggota_id'].toString()),
      bukuId: int.parse(json['buku_id'].toString()),
      namaAnggota: json['nama_anggota'],
      judulBuku: json['judul_buku'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tanggal_pinjam': tanggalPinjam,
      'tanggal_kembali': tanggalKembali,
      'anggota_id': anggotaId,
      'buku_id': bukuId,
    };
  }
} 